framework/db/ActiveQuery.php

```diff
@@ -796,4 +789,89 @@ public function alias($alias)
         }
         return $this;
     }
+
+    /**
+     * Table names based on [[from]]
+     * @return string[] table names
+     * @throws \yii\base\InvalidConfigException
+     * @since 2.0.12
+     */
+    public function getTableNamesUsedInFrom()
+    {
+        $tableNames = [];
+
+        if (empty($this->from)) {
+            $tableNames[] = $this->getPrimaryTableName();
+        } elseif (is_array($this->from)) {
+            $tableNames = array_values($this->from);
+        } elseif (is_string($this->from)) {
+            $tableNames = preg_split('/\s*,\s*/', trim($this->from), -1, PREG_SPLIT_NO_EMPTY);
+        } else {
+            throw new InvalidConfigException(gettype($this->from) . ' in $from is not supported.');
+        }
+
+        // Clean up table names
+        foreach ($tableNames as &$tableName) {
+            $tableName = preg_replace('/^(\w+)\s*.*$/', '$1', $tableName);
+        }
+
+        return $tableNames;
+    }
+
+    /**
+     * @return string[] table aliases based on [[from]]
+     * @throws \yii\base\InvalidConfigException
+     * @since 2.0.12
+     */
+    public function getAliasesUsedInFrom()
+    {
```

Comment:
same here. The question is how durable such naming is for publishing it in the public api. Does it matter if it is called 'getAliases' to show 'what is does' instead of 'how it does it'?